agent.py

```diff
@@ -225,8 +225,12 @@ def run(self, config=None):
         self.collector = Collector(self._agentConfig, emitters, systemStats, hostname)
 
         # In developer mode, the number of runs to be included in a single collector profile
-        self.collector_profile_interval = self._agentConfig.get('collector_profile_interval',
-                                                                DEFAULT_COLLECTOR_PROFILE_INTERVAL)
+        try:
+            self.collector_profile_interval = int(self._agentConfig.get('collector_profile_interval'))
+        except ValueError:
```

Comment:
I think it's a TypeError: `TypeError: int() argument must be a string or a number, not 'NoneType'`.